app/fetchers/space_list_fetcher.rb

```diff
@@ -21,7 +21,11 @@ def filter(message, dataset)
       end
 
       if message.requested? :organization_guids
-        dataset = dataset.where(organization: Organization.where(guid: message.organization_guids))
+        dataset = if message.requested?(:label_selector)
```

Comment:
I think it would be better to not have logic here that depends on another aspect of the message.  If you check out the AppListFetcher or PackageListFetcher, they both just use the map..flatten..map construct no matter what.

Reply:
but we've done that elsewhere.... Not that I agree with it. It was done in the spirit of making the smallest change that worked and not understanding the ORM well enough to improve it